include/envoy/http/filter.h

```diff
@@ -225,6 +225,15 @@ class StreamFilterCallbacks {
    */
   virtual Router::RouteConstSharedPtr route(const Router::RouteCallback& cb) PURE;
 
+  /**
+   * Sets the cached route for the current request to the passed-in RouteConstSharedPtr parameter.
+   *
+   * NOTE: Similar to route(const Router::RouteCallback& cb), this route that is set will be
+   * overridden by clearRouteCache() in subsequent filters. Usage is intended for filters at the end
+   * of the filter chain.
+   */
+  virtual void setRoute(Router::RouteConstSharedPtr route) PURE;
```

Comment:
Let's document what the implication of passing nullptr here is

Reply:
done, see https://github.com/envoyproxy/envoy/pull/15266/commits/9012f546faaa951334f11ac18d76406b62d1bcae